stateMutators.js

```diff
@@ -0,0 +1,212 @@
+const gameState = {
+    basket: {
+        basketValue: 0,
+        xPos: 50,
+        yPos: 10,
+        width: 30,
+        height: 10
+    },
+    score: 0,
+    livesRemaining: 3,
+    gameLevel: 1,
+    gameMode: 'running',
+    fallingObjects: [],
+}
+
+const setBasketPos = (state, basket, value) => {
+    return {
+        ...state,
+        basket: {
+            ...basket,
+            xPos: value
+        }
+    }
+};
+
+const moveBasketLeft = (state, basket, value) => {
+    return {
+        ...state,
+        basket: {
+            ...basket,
+            xPos: basket.xPos - value,
+        },
+    }
+};
+
+const moveBasketRight = (state, basket, value) => {
+    return {
+        ...state,
+        basket: {
+            ...basket,
+            xPos: basket.xPos + value,
+        },
+    }
+};
+
+const setBasketValue = (state, basket, value) => {
+    return {
+        ...state,
+        basket: {
+            ...basket,
+            basketValue: value,
+        },
+    }
+};
+
+const resetBasketValue = (state, basket) => {
+    return {
+        ...state,
+        basket: {
+            ...basket,
+            basketValue: 0,
+        },
+    }
+};
+
+const setScore = (state, value) => {
+    return {
+        ...state,
+        score: value,
+    }
+};
+
+const resetScore = (state) => {
+    return {
+        ...state,
+        score: 0,
+    }
+};
+
+const loseLife = (state, value) => {
```

Comment:
The name of this mutator indicates to me that if I call it a life should be deducted. Though it takes in a value and sets the livesRemaining states to that value. So instead I'd remove the `value` param and just set the state to `livesRemaining - 1`

Reply:
maybe this functionality is already covered by the `calcLives` mutator though

Reply:
I removed it, thanks Jon